### PR TITLE
PIN-4123 Added max number for client keys

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -6,6 +6,7 @@ authorization-process {
     user-registry = ${USER_REGISTRY_API_KEY}
   }
   selfcare-product-id = ${SELFCARE_PRODUCT_ID}
+  threshold-keys = ${THRESHOLD_KEYS}
```
Comment (galales) [inline src/main/resources/application.conf]:
```suggestion
  max-keys-per-client = ${MAX_KEYS_PER_CLIENT}
```

File: src/main/resources/application-standalone.conf [hunk 2]
```diff
@@ -28,6 +28,7 @@ authorization-process {
     user-registry = "a_secret_key"
   }
   selfcare-product-id = ${SELFCARE_PRODUCT_ID}
+  threshold-keys = 1000
```
Comment (galales) [inline src/main/resources/application-standalone.conf]:
```suggestion
  max-keys-per-client = 1000
```

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -506,6 +506,12 @@ paths:
             application/problem+json:
               schema:
                 $ref: '#/components/schemas/Problem'
+        '429':
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: 429 is related to the number of http requests per time unit.
In this case in my opinion we can just use a 400

File: src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiHandlers.scala [hunk 4]
```diff
@@ -118,6 +118,7 @@ object ClientApiHandlers extends AkkaResponses {
       case Failure(ex: OrganizationNotAllowedOnClient)       => forbidden(ex, logMessage)
       case Failure(ex: ClientNotFound)                       => notFound(ex, logMessage)
       case Failure(ex: KeysAlreadyExist)                     => conflict(ex, logMessage)
+      case Failure(ex: TheNumberOfkeysExceedMaximumAllowed)  => tooManyRequests(ex, logMessage)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiHandlers.scala]:
```suggestion
      case Failure(ex: TheNumberOfkeysExceedMaximumAllowed)  => badRequest(ex, logMessage)
```

File: src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala [hunk 5]
```diff
@@ -264,13 +265,21 @@ final case class ClientApiServiceImpl(
...
+
     val result: Future[Keys] = for {
       userId         <- getUidFutureUUID(contexts)
       clientUuid     <- clientId.toFutureUUID
       organizationId <- getOrganizationIdFutureUUID(contexts)
       client         <- authorizationManagementService
         .getClient(clientUuid)
         .ensureOr(client => OrganizationNotAllowedOnClient(clientId, client.consumerId))(_.consumerId == organizationId)
+      keys           <- authorizationManagementService
+        .getClientKeys(clientUuid)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala]: Minor formatting
```suggestion
      keys           <- authorizationManagementService.getClientKeys(clientUuid)
```

File: src/main/scala/it/pagopa/interop/authorizationprocess/common/system/ApplicationConfiguration.scala [hunk 6]
```diff
@@ -25,6 +25,8 @@ object ApplicationConfiguration {
 
   val selfcareProductId: String = config.getString("authorization-process.selfcare-product-id")
 
+  val thresholdKeys: Int = config.getInt("authorization-process.threshold-keys")
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/authorizationprocess/common/system/ApplicationConfiguration.scala]:
```suggestion
  val maxKeysPerClient: Int = config.getInt("authorization-process.max-keys-per-client")
```

File: src/main/scala/it/pagopa/interop/authorizationprocess/error/AuthorizationProcessErrors.scala [hunk 7]
```diff
@@ -70,4 +70,10 @@ object AuthorizationProcessErrors {
   final case class EServiceNotFound(eServiceId: UUID)
       extends ComponentError("0019", s"EService ${eServiceId.toString} not found")
 
+  final case class TheNumberOfkeysExceedMaximumAllowed(clientId: UUID, size: Int)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/authorizationprocess/error/AuthorizationProcessErrors.scala]:
```suggestion
  final case class TooManyKeysPerClient(clientId: UUID, size: Int)
```

**Review by galales — COMMENTED** [review]
Error code + Minor renaming

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**